File: designPatterns/Behavioral_patterns/Mediator/javascript/Component1.js
const BaseComponent = require("./BaseComponent");

/**
 * Concrete Components implement various functionality. They don't depend on
 * other components. They also don't depend on any concrete mediator classes.
 */
class Component1 extends BaseComponent {
  doA() {
    console.log("Component 1 does A.");
    this.mediator.notify(this, "A");
  }

  doB() {
    console.log("Component 1 does B.");
    this.mediator.notify(this, "B");
  }
}

module.exports = Component1;

File: designPatterns/Behavioral_patterns/Mediator/python/Component1.py
"""
Concrete Components implement various functionality. They don't depend on other
components. They also don't depend on any concrete mediator classes.
"""
from BaseComponent import BaseComponent


class Component1(BaseComponent):
    def do_a(self):
        print("Component 1 does A.")
        self.mediator.notify(self, "A")

    def do_b(self):
        print("Component 1 does B.")
        self.mediator.notify(self, "B")
